app/controllers/questions_controller.rb

```diff
@@ -55,6 +55,10 @@ def search
     @questions = Question.search(params[:q])
   end
 
+  def by_tags
+    @questions = Question.by_tags(params[:tags_id])
```

Comment:
The reason why we can use question index here is because the questions are neither all questions nor the users questions but questions by a particular set of tags. I think the pagination you mentioned makes sense. Thanks.